src/app/views/workshops-and-meetings/workshops-and-meetings.component.ts

```diff
@@ -1,60 +1,51 @@
-import { seminars } from 'src/data-entries/json/Seminars';
-import { Component} from '@angular/core';
+import { Component } from '@angular/core';
 import { workshops } from 'src/data-entries/json/workshops';
+import { seminars } from 'src/data-entries/json/Seminars';
+import { DomSanitizer, SafeHtml } from '@angular/platform-browser';
 import { ActivatedRoute, Router } from '@angular/router';
 
 @Component({
   selector: 'app-workshops-and-meetings',
   templateUrl: './workshops-and-meetings.component.html',
   styleUrls: ['./workshops-and-meetings.component.scss']
 })
-
 export class WorkshopsAndMeetingsComponent {
-  $=$;
   workshops = workshops;
   seminars = seminars;
-  type:any;
-  constructor(private route: ActivatedRoute,private router: Router){
-
-  }
-
-
-
-  ngAfterViewInit(): void {
-    //Called after ngAfterContentInit when the component's view has been initialized. Applies to components only.
-    //Add 'implements AfterViewInit' to the class.
-    this.type = <string>this.router.url.split('/').pop();
-    switch(this.type){
-      case this.type != 'Workshops':
-        this.updateData('Seminars')
-        break;
-      default:
-        this.updateData('Workshops')
-        break;
+  showWorkshops: boolean = true;
+  showSeminars: boolean = false;
+
+  constructor(
+    private sanitizer: DomSanitizer,
+    private router: Router,
+    private route: ActivatedRoute
+  ) {
+    // Get the current route URL and set the visibility accordingly
+    const currentRoute = this.router.url.split('/')[1];
+    if (currentRoute === 'Seminars') {
+      this.showWorkshops = false;
+      this.showSeminars = true;
     }
   }
 
-   updateData(dataType:string){
-
-      if(dataType === 'Workshops' ||  dataType === 'Meetings'){
-        $('#Workshops').removeClass('d-none')
-      $('#Seminars').addClass('d-none')
-    }
-    else{
-      $('#Workshops').addClass('d-none')
-      $('#Seminars').removeClass('d-none')
-    }
-
-    $('ol>button').each(function () {
-      if(this.innerHTML !== dataType)
-        $(this).removeClass('font-highlight')
-      else
-        $(this).addClass('font-highlight')
-    });
+  // Sanitize the HTML content before binding it
+  getSanitizedDetails(details: string): SafeHtml {
+    return this.sanitizer.bypassSecurityTrustHtml(details);
```

Comment:
Understand this first and then draft me an explanation why you have done this.

Reply:
You do not understand what you are doing. Please do not copy paste without understanding what it does.
First understand this line " this.sanitizer.bypassSecurityTrustHtml(details);". You have bypassed the checks .
Please think before you answer the next time

Reply:
getSanitizedDetails(details: string): SafeHtml | null {
    if (details) {
      return this.sanitizer.sanitize(SecurityContext.HTML, details) || null;
    }
    return null;
  }
  
  fixed it by using the above code.
  
  Sorry for the oversight. Ofcourse It was new thing to me, but I'm not totally wrong here, I'm quoting the explanation I got, because of that only I put it there. 

"
// Use DomSanitizer's bypassSecurityTrustHtml method to sanitize the HTML content

    return this.sanitizer.bypassSecurityTrustHtml(details);

    // This method returns a SafeHtml object, which represents the sanitized HTML content
    // It tells Angular that the HTML content is safe to render
"